CMakeLists.txt

```diff
@@ -1,5 +1,8 @@
 cmake_minimum_required(VERSION 3.7.2)
 
+# customize the compiler options CMake uses to initialize variables with
+set(CMAKE_USER_MAKE_RULES_OVERRIDE "${CMAKE_CURRENT_SOURCE_DIR}/cmake/CompilerOptionsOverride.cmake")
+
```

Comment:
What's the purpose of this variable and the corresponded file added? I read through Kitware's [documentation](https://cmake.org/cmake/help/v3.8/variable/CMAKE_USER_MAKE_RULES_OVERRIDE.html) of it and still can't understand how SFML is benefiting from it. For what it's worth CI still passes even if you remove this.

Reply:
@binary1248 I see you added this in a759ccd5d. Can you help us understand what purpose it serves?

Reply:
Without it MSVC would complain about duplicate warning flags because a warning flag is already part of the initial set of flags set by CMake.